src/hazardous/mac/poly1305.rs

```diff
@@ -143,93 +148,29 @@ impl core::fmt::Debug for Poly1305 {
 }
 
 impl Poly1305 {
-    #[rustfmt::skip]
-    #[allow(clippy::cast_lossless)]
-    #[allow(clippy::identity_op)]
-    #[allow(clippy::unreadable_literal)]
-    #[allow(clippy::assign_op_pattern)]
+    /// Prime 2^130-5 in little-endian.
+    const PRIME: [u8; 17] = [
+        251, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 3,
+    ];
+
     /// Process a datablock of `POLY1305_BLOCKSIZE` length.
     fn process_block(&mut self, data: &[u8]) -> Result<(), UnknownCryptoError> {
         if data.len() != POLY1305_BLOCKSIZE {
             return Err(UnknownCryptoError);
         }
 
-        let hibit: u32 = if self.is_finalized {
-            0
-        } else {
-            1 << 24
-        };
+        let mut mb = [0u8; 17];
+        mb[..16].copy_from_slice(data);
+        // One byte is appended to detect trailing zeroes if not last chunk.
```

Comment:
```rust
// One byte is appended to detect trailing zeroes if not last chunk.
```

Is this from a Poly1305 standard somewhere? If it is, maybe we can point to a specific section from that standard in this comment. I think if something is surprising enough to warrant a comment, someone might want to go read through the spec for context.

Reply:
Linked to the original Poly1305 design by DJB.

Reply:
That's perfect 👍